lib/howitzer/web/page_dsl.rb

```diff
@@ -0,0 +1,40 @@
+module Howitzer
+  module Web
+    # This module combines page dsl methods
+    module PageDsl
+      # This class is for private usage only
+      class PageScope
+        include RSpec::Matchers
+
+        def initialize(page_klass, &block)
+          self.page_klass = page_klass
+          instance_eval(&block) if block_given?
+        end
+
+        def expected?
+          expect(page_klass.given)
+        end
+
+        def method_missing(name, *args, &block)
+          return super if name =~ /^be_/ || name =~ /^have_/
+          page_klass.given.send(name, *args, &block)
+        end
+
+        private
+
+        attr_accessor :page_klass
+      end
+      # This class is for self usage
+      class ParentPage
+        def self.open(&block)
```

Comment:
This implementation will not open anything :) We need to think how to handle it, I will provide some example later